### DISCUSSION TITLE
Use of Cloud Multicast Repeater

### DISCUSSION THREAD

**Discussion by smnrgrs** [OpenDDS/OpenDDS#4383] [Q&A] [answered]
This question is about how to use the Cloud Multicast Repeater to allow two participants to communicate in the Cloud.

I have two AWS cloud Ubuntu instances A & B in the same VPC/subnet. Firewalls are configured to allow all traffic between them.

I have one simple test participant on each server (domain 42), and my ini file is configured with transport_type=rtps_udp

As multicast traffic cannot pass between the two instances I'm running pairs of repeater.js as follows (based on the multicast groups netcat shows the participants are using):

on A: 
./repeater.js --group 239.255.0.1:17900 --send 15.27.9.6:6180 --uport 6180
on B:
./repeater.js --group 239.255.0.1:17900 --send 15.27.9.11:6180 --uport 6180

on A:
./repeater.js --group 239.255.0.1:17902 --send 15.27.9.6:6181 --uport 6181
on B: 
./repeater.js --group 239.255.0.1:17902 --send 15.27.9.11:6181 --uport 6181

on A:
./repeater.js --group 239.255.0.2:7401  --send 15.27.9.6:6182 --uport 6182
on B:
./repeater.js --group 239.255.0.2:7401  --send 15.27.9.11:6182 --uport 6182

tcpdump shows that the repeaters are transferring the multicast packets between the servers.

The participants don't discover each other. If I add a SpdpSendAddrs line specifying IPs, they do discover each other. 

rtps.ini from server A:
```
[common]
DCPSGlobalTransportConfig=$file
DCPSDefaultDiscovery=TheRTPSConfig
DCPSPendingTimeout=2
DCPSTransportDebugLevel=3
DCPSDebugLevel=3

[rtps_discovery/TheRTPSConfig]
TTL=16
SedpLocalAddress=15.27.9.11:
SpdpLocalAddress=15.27.9.11
SedpAdvertisedLocalAddress=15.27.9.11:

[transport/the_rtps_transport]
transport_type=rtps_udp
TTL=16
local_address=15.27.9.11:
multicast_interface=eth0

```
Does anyone have any idea what I might be missing?

Thanks

**Comment by mitza-oci** [comment]
Is this helpful? https://objectcomputing.com/resources/publications/sett/march-2019-multicast-to-cloud-for-dds-applications

In a two-participant setup I'm not sure running three repeaters is going to provide any advantage.  A single repeater relaying the SPDP messages should be sufficient to bootstrap communication over unicast.

**Reply by smnrgrs** [reply]
Thanks, that is helpful but CheckSourceIp=0 is now also required.

**Comment by jrw972** [comment, marked as answer]
Use only one multicast repeater for SPDP and turn off multicast for SEDP and the transport.  You should be able to remove the TTL and IPs as well.  I think you need to set `CheckSourceIp=0` to prevent the SPDP message from the repeater from being dropped because the source IP and locators will not agree. 

```
[common]
DCPSGlobalTransportConfig=$file
DCPSDefaultDiscovery=TheRTPSConfig
DCPSPendingTimeout=2
DCPSTransportDebugLevel=3
DCPSDebugLevel=3

[rtps_discovery/TheRTPSConfig]
CheckSourceIp=0
SedpMulticast=0

[transport/the_rtps_transport]
transport_type=rtps_udp
use_multicast=0
```

**Reply by smnrgrs** [reply]
Thanks, CheckSourceIp=0 was the magic I was missing!
